core/trino-main/src/main/java/io/trino/sql/planner/plan/LastNNode.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import io.trino.sql.planner.Symbol;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+
+@Immutable
+public class LastNNode
+        extends PlanNode
```

Comment:
i think a new PlanNode subclass needs to be registered in the parent PlanNode class (see json annotation)

Reply:
Yeah, I think I missed that.